### PR TITLE
No subscriptions to AuthGuardWithForcedLogin.

### PR DESCRIPTION
**Pull request by 1304654**
Refactor coding with no subscriber.

### CODE REVIEW

**Comment by jeroenheijmans** [comment]
Thanks @1304654 that does look a bit cleaner. I'll have a test some time later and merge then. 👍

File: src/app/core/auth-guard-with-forced-login.service.ts [hunk 1]
```diff
@@ -7,21 +7,22 @@ import { AuthService } from './auth.service';
...
-    this.authService.isAuthenticated$.subscribe(i => this.isAuthenticated = i);
+
   }
 
   canActivate(
     route: ActivatedRouteSnapshot,
     state: RouterStateSnapshot,
   ): Observable<boolean> {
     return this.authService.isDoneLoading$
+      .pipe(withLatestFrom(this.authService.isAuthenticated$))
```
Comment (jeroenheijmans) [inline src/app/core/auth-guard-with-forced-login.service.ts:23]: This line should be after the `filter` I think (or the filter should also handle an argument that is an array of bools (e.g. `[isDone, _]`).

Comment (jeroenheijmans) [inline src/app/core/auth-guard-with-forced-login.service.ts:23]: The `withLatestFrom` has to be imported, or else it won't compile.

Comment (jeroenheijmans) [inline src/app/core/auth-guard-with-forced-login.service.ts:23]: We could consider doing something like a `switchMap` instead? We only need to check if `isDoneLoading$` is true, and then switch over to `isAuthenticated`.

File: src/app/core/auth-guard-with-forced-login.service.ts [hunk 2]
```diff
@@ -7,21 +7,22 @@ import { AuthService } from './auth.service';
...
       .pipe(filter(isDone => isDone))
-      .pipe(tap(_ => this.isAuthenticated || this.authService.login(state.url)))
-      .pipe(map(_ => this.isAuthenticated));
+      .pipe(tap([_, isAuthenticated] => isAuthenticated || this.authService.login(state.url)))
+      .pipe(map([_, isAuthenticated] => isAuthenticated));
```
Comment (jeroenheijmans) [inline src/app/core/auth-guard-with-forced-login.service.ts:26]: I think that the array needs wrapping in parens, i.e. `tap(([_, isAuth]) => ....`

**Comment by jeroenheijmans** [comment]
Final version (I did the merge locally, didn't push to update the branch): https://github.com/jeroenheijmans/sample-angular-oauth2-oidc-with-auth-guards/commit/36316ee1971a8a8160033f55ba7eabe14f7d3add

**Comment by 1304654** [comment]
It's sounds good. You are righy.

**Pull request merged**